ipamd/datastore/data_store.go

```diff
@@ -213,7 +214,7 @@ func (ds *DataStore) assignPodIPv4AddressUnsafe(k8sPod *k8sapi.K8SPodInfo) (stri
 					addr.address, k8sPod.Name, k8sPod.Namespace)
 				return addr.address, eni.DeviceNumber, nil
 			}
-			if !addr.Assigned && k8sPod.IP == "" {
+			if !addr.Assigned && k8sPod.IP == "" && curTime.Sub(addr.unAssignedTime) > 30*time.Second {
```

Comment:
How is this time related to the `addressCoolingPeriod` const?

Reply:
+1 on Matt's question.  The values seem related, though the defined constant seems to be used to guard when it's ok to free an ENI.  If they are coupled, reuse the defined constant.  If not, make another named constant instead of the inline `30*time.Second`.

I note that `addr.unAssignedTime` may be nil.  In this case `Sub` appears to compute a duration from epoch, but this is accidental, [`not documented`](https://golang.org/pkg/time/#Time.Sub) behavior.  It would be safer for code to handle the nil case explicitly, to protect for this undocumented behavior changing.  I also note the time package has a [`Since`](https://golang.org/pkg/time/#Since) function that would simplify the expression to `Since(addr.unAssignedTime) > addressCoolingPeriod` and save having to set curTime above.

Reply:
Upon deeper look, [time.Time](https://golang.org/pkg/time/#Time) is struct, only one element of which can be nil.  The Zero value of time is defined as the unix epoch, so the behavior is not undocumented.  I still advocate a named constant, commentary, and use of Since in preference to now.Sub(prior).